frame/vesting/src/tests.rs

```diff
@@ -112,6 +208,32 @@ fn vested_balance_should_transfer() {
 		});
 }
 
+#[test]
+fn vested_balance_should_transfer_with_multi_sched() {
+	ExtBuilder::default().existential_deposit(ED).build().execute_with(|| {
+		let sched0 = VestingInfo::new(5 * ED, 128, 0);
+		assert_ok!(Vesting::vested_transfer(Some(13).into(), 1, sched0));
+		// Total of 2560 of locked for all the schedules.
+		assert_eq!(Vesting::vesting(&1).unwrap(), vec![sched0, sched0]);
+
+		let user1_free_balance = Balances::free_balance(&1);
+			assert_eq!(user1_free_balance, 3840); // Account 1 has free balance
+
+			// Account 1 has only 256 units unlocking at block 1 (plus 1280 unvested)
+			assert_eq!(Vesting::vesting_balance(&1), Some(2304));
+			assert_ok!(Vesting::vest(Some(1).into()));
+			assert_ok!(Balances::transfer(Some(1).into(), 2, 1536));
+		});
+}
+
+#[test]
+fn vest_correctly_fails() {
```

Comment:
```suggestion
fn non_vested_cannot_vest() {
```

(just a better name, since `vest` can fail for various reasons.)